### PR TITLE
Update expire_date column of table comments

### PR DESCRIPTION
**Pull request by vitormattos**
The Talk app started to use this column but we didn't implemented the update feature to change the expire date of a specific comment.

We need to have this at server side, as talk is not the only thing using comments and the API should work for all things.

### CODE REVIEW

File: lib/private/Comments/Manager.php [hunk 1]
```diff
@@ -1345,7 +1345,8 @@ protected function updateQuery(IComment $comment, bool $tryWritingReferenceId):
 			->set('creation_timestamp', $qb->createNamedParameter($comment->getCreationDateTime(), 'datetime'))
 			->set('latest_child_timestamp', $qb->createNamedParameter($comment->getLatestChildDateTime(), 'datetime'))
 			->set('object_type', $qb->createNamedParameter($comment->getObjectType()))
-			->set('object_id', $qb->createNamedParameter($comment->getObjectId()));
+			->set('object_id', $qb->createNamedParameter($comment->getObjectId()))
+			->set('expire_date', $qb->createNamedParameter($comment->getExpireDate(), 'datetime'));
```
Comment (nickvergessen) [inline lib/private/Comments/Manager.php:1349]: does this work with `null`?

Comment (vitormattos) [reply lib/private/Comments/Manager.php:1349]: I created the PR with developing label to run all the CI to check this. But... I think that we will need to test this.

Comment (vitormattos) [reply lib/private/Comments/Manager.php:1349]: Workded fine. The field expire_date is nullable.
https://github.com/nextcloud/server/blob/master/core/Migrations/Version25000Date20220602190540.php#L49-L51

Comment (nickvergessen) [reply lib/private/Comments/Manager.php:1349]: Then add a test were it is being updated?

**Comment by nickvergessen** [comment]
> I think that we can close this PR and don't merge.

We should fix it, as talk is not the only thing using comments and the API should work for all things.
If tests exist for the comments manager please also add a test for this case

File: lib/private/Comments/Manager.php (hunk 1 above)
Comment (vitormattos) [reply lib/private/Comments/Manager.php:1355]: Done.

**Review by nickvergessen — APPROVED** [review]

**Comment by vitormattos** [comment]
Description updated.

**Review by Pytal — APPROVED** [review]

**Pull request merged**